electrolytes/solvation/solvate.py

```diff
@@ -0,0 +1,545 @@
+            xyz_paths.extend(crawl_xyz(item.resolve()))
+    return xyz_paths
+
+
+def dump_xyzs(
+    complexes: List[Tuple[Atoms, int, int]],
+    prefix: str,
+    base_dir: Path
+):
+    """
+    Create *.xyz files for each complex in a set of solvated complexes
+
+    Args:
+        complexes (List[Tuple[Atoms, int, int]]): Collection of solvated molecules. Each entry is a molecular
+            structure, its charge, and its spin multiplicity
+        prefix (str): Prefix for all *.xyz files
+        base_dir (str | Path): Path in which to dump *.xyz files
+
+    Returns:
+        None
+    """
+
+    base_dir.mkdir(exist_ok=True)
+
+    for ii, (atoms, charge, spin) in enumerate(complexes):
+        write((base_dir / f"{prefix}_solv{ii}_{charge}_{spin}.xyz"), atoms, format="xyz")
+
+
+if __name__ == "__main__":
+    
+    # For reproducibility
+    random.seed(42)
+
+    # TODO: CHANGE THESE
+    # `xyz_dir` should point to a root-level directory with *.xyz files
+    # *.xyz's can be nested multiple levels down - this script will recursively search for them
+    xyz_dir = Path("xyzs")
+    # `base_dir` should point to root-level directory where generated complexes will be dumped
+    base_dir = Path("dump")
+    ood_dir = Path("ood")
+
+    base_dir.mkdir(exist_ok=True)
+    ood_dir.mkdir(exist_ok=True)
+
+    # Set-up: get info from predefined set of SMILES
+    solvating_info = info_from_smiles(
+        metals + other_cations + anions + neutrals
+    )
+
+    just_solvent_info = info_from_smiles(
+        neutrals
+    )
+
+    # Identify all molecules for solvation
+    # NOTE: it's important the files have the format "<NAME>_<charge>_<spin>.xyz(.gz)"
+    # Because otherwise, we can't identify charge/spin information from the *.xyz format
+    xyz_files = crawl_xyz(xyz_dir)
+    print("TOTAL NUMBER OF XYZ FILES:", len(xyz_files))
+
+    # For now (2024/06/13), the plan is to do the following for each molecule:
+    # 1. Generate `m` random dimers with neutral molecules or molecules of opposite charge
+    # 2. Generate 1 solvation shell with a random pure solvent from a pre-defined list
+    # 3. Generate `n` random solvation shells with combinations of random combinations of components (ions,
+    #    neutral species, etc.)
+
+    # TODO: CHANGE THESE
+    # These numbers just for testing
+    num_dimers = 3
+    num_random_shells = 1
+    max_core_molecule_size = 50
+    max_atom_budget = 65
+
+    # TODO: play around with these more
+    # In initial testing, random placement seems to help better surround central molecule
+    # Sella might be helpful, but also really slows things down
+    architector_params={"species_location_method": "random"}
+
+    for xyz_file in xyz_files:
+
+        mol = Molecule.from_file(xyz_file)
+
+        # If molecule is too large, don't bother trying to make solvation complexes
+        if len(mol) > max_core_molecule_size:
+            continue
+
+        name = xyz_file.name.split(".")[0]
```

Comment:
```suggestion
        name = os.path.splitext(xyz_file.name)[0]
```
Or without `.name` if using glob.